src/util/duration.h

```diff
@@ -23,43 +23,43 @@ class DurationBase {
 
     // Returns the duration as an integer number of seconds (rounded-down).
     constexpr qint64 toIntegerSeconds() const {
-        return m_durationNanos / kNanosPerSecond;
+        return static_cast<quint64>(m_durationNanos) / kNanosPerSecond;
```

Comment:
`return static_cast<quint64>(toDoubleSeconds())`?

The comment "(rounded-down)" could be misleading unless we prevent negative durations.

Reply:
It looks like this is dead code anyway. Can we remove it?

Reply:
It's true that this method is only used by the class itself and by tests:

```
$ grep -HrnF toIntegerSeconds ../src
../src/test/duration_test.cpp:13:    EXPECT_EQ(0, d.toIntegerSeconds());
../src/test/duration_test.cpp:19:    EXPECT_EQ(1, d.toIntegerSeconds());
../src/test/duration_test.cpp:27:    EXPECT_EQ(0, d.toIntegerSeconds());
../src/test/duration_test.cpp:33:    EXPECT_EQ(1e3, d.toIntegerSeconds());
../src/test/duration_test.cpp:41:    EXPECT_EQ(0, d.toIntegerSeconds());
../src/test/duration_test.cpp:47:    EXPECT_EQ(1e6, d.toIntegerSeconds());
../src/test/duration_test.cpp:55:    EXPECT_EQ(255, d.toIntegerSeconds());
../src/test/duration_test.cpp:61:    EXPECT_EQ(1e9, d.toIntegerSeconds());
../src/test/duration_test.cpp:69:    EXPECT_EQ(5, d.toIntegerSeconds());
../src/test/duration_test.cpp:70:    EXPECT_EQ(2, d2.toIntegerSeconds());
../src/test/duration_test.cpp:71:    EXPECT_EQ(7, d3.toIntegerSeconds());
../src/test/duration_test.cpp:79:    EXPECT_EQ(7, d.toIntegerSeconds());
../src/test/duration_test.cpp:80:    EXPECT_EQ(2, d2.toIntegerSeconds());
../src/test/duration_test.cpp:88:    EXPECT_EQ(5, d.toIntegerSeconds());
../src/test/duration_test.cpp:89:    EXPECT_EQ(2, d2.toIntegerSeconds());
../src/test/duration_test.cpp:90:    EXPECT_EQ(3, d3.toIntegerSeconds());
../src/test/duration_test.cpp:98:    EXPECT_EQ(-5, d.toIntegerSeconds());
../src/test/duration_test.cpp:99:    EXPECT_EQ(10, d2.toIntegerSeconds());
../src/util/duration.h:25:    constexpr qint64 toIntegerSeconds() const {
../src/util/duration.h:115:            return debug << dd.toIntegerSeconds() << "s";
../src/util/duration.h:255:        return QString("%1 s").arg(toIntegerSeconds());
```

However, the other `toIntegerFoo()` methods are used elsewhere, and it would be weird that such a method is only missing for a single unit, but provided for others.